test: Use fake profile data instead of test_usa_tamu

c-voegele opened the pull request:
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)



### Purpose
Hi Breakthrough Energy Team! Thank you for your work on this interesting and important project!

This PR is an attempt to fix https://github.com/Breakthrough-Energy/PowerSimData/issues/489.

The  `test_usa_tamu` profiles are no longer needed for `test_transform_profile` and no Internet connection is required to run the tests.

### What the code is doing
To avoid the Azure blob call, I mocked `InputData` to return random profiles. I just used uniform random data in [0, 1) as suggested [here](https://github.com/Breakthrough-Energy/PowerSimData/pull/485#issuecomment-848331723)

### Testing
I ran `tox -e pytest-local -- --cov=powersimdata` and `tox -e checkformatting` and both are passing.

To make sure that `test_usa_tamu` was not required, I wiped `~/ScenarioData` and rebuilt it with `LocalConfig().initialize()`. After running the tests, `test_usa_tamu` is no longer loaded.

### Where to look
`powersimdata/input/tests/test_transform_grid.py`

### Usage Example/Visuals
n/a 

### Time estimate
15 minutes

danielolsen commented on powersimdata/input/tests/test_transform_profile.py:
```diff
@@ -36,6 +34,11 @@ def get_plant_with_resource(base_grid, resource):
 
 
+def _get_plant_ids_for_type(grid, type):
+    plant_ids = list(grid.plant.isin(profile_type[type]).query("type == True").index)
```
I think `grid.plant.isin(profile_type[type]).query("type == True")` can be simplified to `grid.plant.query("type in @profile_type[@type]")`.

c-voegele replied:
Done, thanks!

BainanXia commented:
This is nice! I'm wondering whether it is possible to pull `mock_input_data` and corresponding raw profile functions into a separate module within `powersimdata/tests/` together with other mocks so that it could potentially be used elsewhere, then we could import and patch it in `test_transform_profile.py`.

BainanXia commented:
@jon-hagg With the `mock_input_data` implemented, do we still need truncated test profiles in blob storage under `test_usa_tamu` folder?

jon-hagg commented:
> @jon-hagg With the `mock_input_data` implemented, do we still need truncated test profiles in blob storage under `test_usa_tamu` folder?

I would keep them for now - they are used [here](https://github.com/Breakthrough-Energy/PowerSimData/blob/develop/powersimdata/scenario/tests/test_scenario.py#L63) to test scenario preparation

c-voegele commented:
> This is nice! I'm wondering whether it is possible to pull `mock_input_data` and corresponding raw profile functions into a separate module within `powersimdata/tests/` together with other mocks so that it could potentially be used elsewhere, then we could import and patch it in `test_transform_profile.py`.

Thanks for your suggestion @BainanXia! I reorganized the code as you suggested. Let me know what you think!

BainanXia commented on powersimdata/tests/mock_input_data.py:
```diff
@@ -0,0 +1,67 @@
+from powersimdata.input.grid import Grid
+
+
+class MockInputData:
```
It would be great if we can have some doc strings here given it is now in a user facing `tests` module.

c-voegele replied:
Sure, added docs and tests!

BainanXia commented on powersimdata/tests/mock_input_data.py:
```diff
@@ -0,0 +1,67 @@
+        self,
+        grid: Grid,
+        start_time="2016-01-01 00:00:00",
+        periods=24,
```
Given this mock class can potentially be used in a general sense, maybe we can make this flexible by specifying the `end_time` or we could keep `periods` argument as well just in case people don't want to bother.

c-voegele replied:
Done!

BainanXia commented:
> > This is nice! I'm wondering whether it is possible to pull `mock_input_data` and corresponding raw profile functions into a separate module within `powersimdata/tests/` together with other mocks so that it could potentially be used elsewhere, then we could import and patch it in `test_transform_profile.py`.
> 
> Thanks for your suggestion @BainanXia! I reorganized the code as you suggested. Let me know what you think!

Thanks for the contribution! I think we are good to go except for some minor concerns above.

c-voegele commented:
> > > This is nice! I'm wondering whether it is possible to pull `mock_input_data` and corresponding raw profile functions into a separate module within `powersimdata/tests/` together with other mocks so that it could potentially be used elsewhere, then we could import and patch it in `test_transform_profile.py`.
> > 
> > 
> > Thanks for your suggestion @BainanXia! I reorganized the code as you suggested. Let me know what you think!
> 
> Thanks for the contribution! I think we are good to go except for some minor concerns above.

Thanks! I addressed your comments. Let me know if you find anything else!

BainanXia commented on powersimdata/input/tests/test_transform_profile.py:
```diff
@@ -197,39 +191,42 @@ def base_grid():
-    }
-    profile = input_data.get_data(profile_info, kind)
-    return profile_info, profile
+@pytest.fixture(scope="module")
+def input_data(base_grid):
+    mock_input_data = MockInputData(base_grid)
+    return mock_input_data
+
+
+@pytest.fixture(scope="module", autouse=True)
+def mock_input_data_class(input_data):
+    with patch(
+        "powersimdata.input.transform_profile.InputData"
+    ) as mock_input_data_class:
+        mock_input_data_class.return_value = input_data
+        yield
```
This is clean.

BainanXia reviewed (approved):
Great! Thanks for improving test coverage and the patience @c-voegele

BainanXia commented:
> Thanks for the approval!
> 
> It doesn't look like I have write permissions so this is as far as I can go.

No worries, we will take care of it.
